### PR TITLE
fix firefox-container / chrome-container browser modes

### PR DESCRIPTION
**Pull request by jtnord**
ammends #1100 even with the fix of the port flag selenium would not start due to

```
11:43:39.232 INFO [LocalDistributor.newSession] - Session request received by the Distributor:
 [Capabilities {acceptInsecureCerts: true, browserName: firefox, moz:debuggerAddress: true, moz:firefoxOptions: {prefs: {dom.disable_beforeunload: false, dom.max_chrome_script_run_time: 600000, dom.max_script_run_time: 600000, intl.accept_languages: en}}, proxy: {httpProxy: localhost:33435, proxyType: manual, sslProxy: localhost:33435}}, Capabilities {acceptInsecureCerts: true, browserName: firefox, moz:debuggerAddress: true, moz:firefoxOptions: {prefs: {dom.disable_beforeunload: false, dom.max_chrome_script_run_time: 600000, dom.max_script_run_time: 600000, intl.accept_languages: en}}, proxy: {httpProxy: localhost:33435, proxyType: MANUAL, sslProxy: localhost:33435}}]
1680781419259   geckodriver     INFO    Listening on 127.0.0.1:21862
1680781419500   mozrunner::runner       INFO    Running command: "/usr/bin/firefox" "--marionette" "--remote-debugging-port" "6734" "--remote-allow-hosts" "localhost" "--remote-allow-origins" "http://127.0.0.1:6734/,http://localhost:6734/,http://[::1]:6734/" "-no-remote" "-profile" "/tmp/rust_mozprofilebyn9Y6"
Unable to init server: Could not connect: Connection refused
Error: cannot open display: :17
11:43:39.626 WARN [DriverServiceSessionFactory.apply] - Error while creating session with the driver service. Stopping driver service: Could not start a new session. Response code 500. Message: Process unexpectedly closed with status 1
Host info: host: 'debian', ip: '127.0.1.1'
Build info: version: '4.6.0', revision: '79f1c02ae20'
```

more variables need to be set to configure the display xvfb and vnc. reverse engineered from the docker file and tested in a linux (code requires host based networking for docker to work) VM with

```
export JAVA_OPTS=-Djava.awt.headless=true
export JENKINS_JAVA_OPTS="-Xmx1280m -Djava.awt.headless=true" export JENKINS_WAR=target/jenkins-war.war
export BROWSER=firefox-container
mvn test -Dtest=CreateItemTest
```

fixes:#1090

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/org/jenkinsci/test/acceptance/FallbackConfig.java [hunk 1]
```diff
@@ -233,7 +233,9 @@ private WebDriver createContainerWebDriver(TestCleaner cleaner, String image, Mu
             String[] args = {
                     "run", "-d", "--shm-size=2g", "--network=host",
                     "-e", "SE_OPTS=--port " + controlPort,
-                    "-e", "DISPLAY=:" + displayNumber,
+                    "-e", "DISPLAY=:" + displayNumber + ".0",
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/test/acceptance/FallbackConfig.java:236]: see https://hub.docker.com/layers/selenium/standalone-firefox/4.6.0/images/sha256-1714c76e1da1badf6e26b754a2293213c352e53bf92e4f4729c3d3c8743dbd75?context=explore  lines 50-52

**Review by uhafner — APPROVED** [review]

**Comment by uhafner** [comment]
This still seems not to work, I upgraded to the lastest ATH version: https://github.com/jenkinsci/warnings-ng-plugin/pull/1470

Can you run any ATH tests locally with `BROWSER=chrome-container`? 

I always get:
```
Caused by: java.io.UncheckedIOException: java.net.ConnectException: Connection refused: /127.0.0.1:56878
```

**Comment by jtnord** [comment]
> Caused by: java.io.UncheckedIOException: java.net.ConnectException: Connection refused: /127.0.0.1:56878

Is that the chrome failing to talk to Jenkins, the HAR capturing proxy, Selenium trying to talk to webdriver?
And from What OS?  I would only expect that to work on Linux.

Please create a new issue with full details.

**Comment by uhafner** [comment]
The issue is already there: #1090

In https://github.com/jenkinsci/warnings-ng-plugin/pull/1470 you will find all errors for Linux.

I tried on MacOS as well. Do you successfully used `BROWSER=chrome-container` on any configuration?

**Pull request merged**